core/src/main/java/org/elasticsearch/search/sort/FieldSortBuilder.java

```diff
@@ -129,31 +182,188 @@ public FieldSortBuilder setNestedPath(String nestedPath) {
         return this;
     }
 
+    /** Returns the nested path if sorting occurs in a field that is inside a nested object. */
+    public String getNestedPath() {
+        return this.nestedPath;
+    }
+
     @Override
     public XContentBuilder toXContent(XContentBuilder builder, Params params) throws IOException {
         builder.startObject(fieldName);
         if (order != null) {
-            builder.field("order", order.toString());
+            builder.field(ORDER.getPreferredName(), order.toString());
         }
         if (missing != null) {
-            builder.field("missing", missing);
-        }
-        if (ignoreUnmapped != null) {
-            builder.field(SortParseElement.IGNORE_UNMAPPED.getPreferredName(), ignoreUnmapped);
+            if (missing instanceof BytesRef) {
+                builder.field(MISSING.getPreferredName(), ((BytesRef) missing).utf8ToString());
+            } else {
+                builder.field(MISSING.getPreferredName(), missing);
+            }
         }
         if (unmappedType != null) {
-            builder.field(SortParseElement.UNMAPPED_TYPE.getPreferredName(), unmappedType);
+            builder.field(UNMAPPED_TYPE.getPreferredName(), unmappedType);
         }
         if (sortMode != null) {
-            builder.field("mode", sortMode);
+            builder.field(SORT_MODE.getPreferredName(), sortMode);
         }
         if (nestedFilter != null) {
-            builder.field("nested_filter", nestedFilter, params);
+            builder.field(NESTED_FILTER.getPreferredName(), nestedFilter, params);
         }
         if (nestedPath != null) {
-            builder.field("nested_path", nestedPath);
+            builder.field(NESTED_PATH.getPreferredName(), nestedPath);
         }
         builder.endObject();
         return builder;
     }
+
+    @Override
+    public boolean equals(Object other) {
```

Comment:
Why not also test for object reference equality here like we do in AbstractQueryBuilder?